examples/mi-mctp.c

```diff
@@ -79,31 +80,17 @@ static int show_port(nvme_mi_ep_t ep, int portid)
 	return 0;
 }
 
-int main(int argc, char **argv)
+int do_info(nvme_mi_ep_t ep)
 {
 	struct nvme_mi_nvm_ss_health_status ss_health;
 	struct nvme_mi_read_nvm_ss_info ss_info;
-	nvme_mi_ep_t ep;
-	uint8_t eid;
-	int net;
-	int rc;
-	int i;
-
-	if (argc != 3) {
-		fprintf(stderr, "usage: %s <net> <eid>\n", argv[0]);
-		return EXIT_FAILURE;
-	}
-
-	net = atoi(argv[1]);
-	eid = atoi(argv[2]) & 0xff;
-
-	ep = nvme_mi_open_mctp(net, eid);
-	if (!ep)
-		err(EXIT_FAILURE, "can't open MCTP endpoint %d:%d", net, eid);
+	int i, rc;
 
 	rc = nvme_mi_mi_read_mi_data_subsys(ep, &ss_info);
-	if (rc)
-		err(EXIT_FAILURE, "can't perform Read MI Data operation");
+	if (rc) {
+		warn("can't perform Read MI Data operation");
+		return -1;
```

Comment:
return EXIT_FAILURE

I just mentioning it once, there are couple of more places later in the series which do the same.

Reply:
Even though this isn't an exit status?

Reply:
You are right. Although returning a ESOMETING would be better IMO in this particular case.

Reply:
ok, I'll figure out an appropriate value for these.